presto-hive/src/test/java/io/prestosql/plugin/hive/metastore/glue/TestHiveGlueMetastore.java

```diff
@@ -152,34 +154,59 @@ public void testRenameTable()
         // rename table is not yet supported by Glue
     }
 
-    @Override
-    public void testPartitionStatisticsSampling()
-    {
-        // Glue metastore does not support column level statistics
-    }
-
     @Override
     public void testUpdateTableColumnStatistics()
+            throws Exception
     {
-        // column statistics are not supported by Glue
+        // The original implementation AbstractTestHiveClient#testUpdateTableColumnStatistics assumes each update call to metastore would override
+        // the previous update call but this is not entirely true for Glue.
+        // For example, there are 5 columns of a table. if you update 2 columns and then update the remaining 3 columns in two separate calls.
+        // The second update call would not override the first update call (assume the first 2 and remaining 3 columns are exclusive).
+        // Following this rationale, provide an empty column stat does not clear all the existing column stats.
+        // (Glue actually does not accept the empty column stat).
+        SchemaTableName tableName = temporaryTable("update_table_column_statistics");
+        try {
+            doCreateEmptyTable(tableName, ORC, STATISTICS_TABLE_COLUMNS);
+            // STATISTICS_1_1 must be subset of STATISTICS_1
+            testUpdateTableStatistics(tableName, EMPTY_TABLE_STATISTICS, STATISTICS_1_1, STATISTICS_1);
+        }
+        finally {
+            dropTable(tableName);
+        }
     }
 
     @Override
     public void testUpdateTableColumnStatisticsEmptyOptionalFields()
+            throws Exception
     {
-        // column statistics are not supported by Glue
+        // this test is not really meaningful for Glue, for numeric columns, min/max stat should always be available
+        // The only exception is when all rows are NULLs. Glue requires min/max to be set.
```

Comment:
Can we then change this to an `assertThat(super.testUpdateTableColumnStatisticsEmptyOptionalFields()).throws(xxx)`?

That way if Glue's behaviour changes in future we'll know that we can/need to change our impl.

Same for other commented out tests.

Reply:
Agree, but we should also be careful with this since glue is returning http status code `500` so this case is not handled by their api yet, I've added some specific check on the error message to address this